package/tvheadend-app-osmc/files/DEBIAN/postinst

```diff
@@ -1,33 +1,86 @@
-#!/bin/bash
+#! /bin/bash
 
 if [ "$1" = "configure" ]; then
 
 	systemctl daemon-reload
 
-	if [ ! -f /home/osmc/.hts/tvheadend/accesscontrol/1 ] && [ -e "/var/run/${DPKG_MAINTSCRIPT_PACKAGE}_install" ]; then
+	# check for existing user called osmc
+	OSMCUSERFILE=$(grep -l "\"username\": \"osmc\"" /home/osmc/.hts/tvheadend/accesscontrol/* )
+	echo userfile is: $OSMCUSERFILE
 
-		echo "Creating TVHeadend default access control file"
-		mkdir -p /home/osmc/.hts/tvheadend/accesscontrol
+	if [ -z "$OSMCUSERFILE" ]; then
```

Comment:
Another check if useful, is the ability to check which version you're upgrading from.

if $(dpkg --compare-versions "$2" lt "x.x.x"); then ....

This might make migration easier across future versions.